packages/tools/viewer-alpha/src/viewerElement.ts

```diff
@@ -215,6 +216,35 @@ export class HTML3DElement extends LitElement {
         return this._viewerDetails;
     }
 
+    /**
+     * Get hotspot world and canvas values from a named hotspot
+     * @param slot slot of the hot spot
+     * @param res resulting canvas and world positions
+     * @returns world and canvas space coordinates
+     */
+    public queryHotSpot(slot: string, res: HotSpotPositions): boolean {
+        // Retrieve all hotspots inside the viewer element
+        const hotspots = this.querySelectorAll(".hotspot");
+        const hotSpotQuery = new HotSpotQuery();
+        let resultFound = false;
+        // Iterate through each hotspot to get the 'data-surface' and 'data-name' attributes
+        hotspots.forEach((hotspot) => {
+            const slotAttribute = hotspot.getAttribute("slot");
+            if (slot === slotAttribute) {
+                const dataSurface = hotspot.getAttribute("data-surface");
+                const array = dataSurface!.split(" ");
+                const scene = this._viewerDetails?.scene;
+                const mesh = scene?.getMeshByName(array[0]);

```

Comment:
So this is using mesh name to specify the target mesh? I assume mesh name is not unique, would it make more sense to stick with index (that is, index within the meshes array of the asset container)?

Reply:
Both solution have pros and cons. I've reverted to an index.